scripts/fig_2b-qc_plots.R

```diff
@@ -0,0 +1,278 @@
+        axis.ticks = element_blank(),
+        axis.text = element_blank(),
+        panel.background = element_rect(colour = "black", linewidth = 0.5),
+        aspect.ratio = 1)
+
+# HVGs -------------------------------------------------------------------------
+
+# select top genes to plot
+all_hvg <- processed_meta$highly_variable_genes
+# select 4 of the top 5 genes that have short gene names 
+# skip the first one since it doesn't have a gene symbol 
+top_genes <- all_hvg[2:5]
+
+# grab expression for top genes from counts
+var_gene_exp <- logcounts(processed_sce[top_genes, ]) |>
+  as.matrix() |>
+  t() |>
+  as.data.frame() |>
+  tibble::rownames_to_column("barcode")
+
+# grab rowdata as data frame to later combine with gene expression data
+# rowdata contains the mapped gene symbol so can use that to label plots instead of ensembl gene id from rownames
+rowdata_df <- rowData(processed_sce) |>
+  as.data.frame() |>
+  tibble::rownames_to_column("ensembl_id") |>
+  select(ensembl_id, gene_symbol) |>
+  filter(ensembl_id %in% top_genes) |>
+  mutate(
+    gene_symbol = ifelse(!is.na(gene_symbol), gene_symbol, ensembl_id),
+    ensembl_id = factor(ensembl_id, levels = top_genes)
+  ) |>
+  arrange(ensembl_id) |>
+  mutate(gene_symbol = factor(gene_symbol, levels = gene_symbol))
+
+
+# extract umap embeddings as a dataframe to join with gene expression and coldata for plotting
+umap_df <- reducedDim(processed_sce, "UMAP") |>
+  as.data.frame() |>
+  tibble::rownames_to_column("barcode")
+
+# combine gene expression with coldata, umap embeddings, and rowdata and create data frame to use for plotting
+coldata_df <- colData(processed_sce) |>
+  as.data.frame() |>
+  tibble::rownames_to_column("barcode") |>
+  # combine with gene expression
+  left_join(var_gene_exp, by = "barcode") |>
+  # combine with umap embeddings
+  left_join(umap_df, by = "barcode") |>
+  # combine all genes into a single column for easy faceting
+  tidyr::pivot_longer(
+    cols = starts_with("ENSG"),
+    names_to = "ensembl_id",
+    values_to = "gene_expression"
+  ) |>
+  # join with row data to add in gene symbols
+  left_join(rowdata_df)
+
+# faceted plot for hvgs
+hvg_plot <- ggplot(coldata_df, aes(x = UMAP1, y = UMAP2, color = gene_expression)) +
+  geom_point(alpha = 0.1, size = 0.001) +
+  facet_wrap(vars(gene_symbol)) +
+  scale_color_viridis_c() +
+  theme(
+    legend.position = "none",
+    axis.title = element_text(size = 9, color = "black"),
+    strip.text = element_text(size = 8)
+  ) 
+
+# Combine plots ----------------------------------------------------------------
+
+plot_list <- list(knee_plot,
+                  cell_metrics_plot,
+                  miQC_plot,
+                  filtered_plot,
+                  umap_plot, 
+                  hvg_plot)
+patchwork::wrap_plots(plot_list, ncol = 3) &
+  theme(text = element_text(size = 10))
+
+# save files 
+ggsave(output_plot_file, width = 8.5, height = 5.5, units = "in")
```

Comment:
I'd just be explicit here, since it is part of a script.
```suggestion
combined_plot <- patchwork::wrap_plots(plot_list, ncol = 3) &
  theme(text = element_text(size = 10))

# save files 
ggsave(output_plot_file, plot = combined_plot, width = 8.5, height = 5.5, units = "in")
```